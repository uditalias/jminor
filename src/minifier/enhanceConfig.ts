import ICompressConfig from "../interfaces/ICompressConfig";
import * as merge from "deepmerge";

export default function enhanceConfig(config: ICompressConfig = {}): ICompressConfig {
    const defaults = {
        translateKeys: true,
        null: {
            removeNull: false,
            exclude: []
        },
        boolean: {
            removeFalse: false,
            exclude: []
        },
        string: {
            removeEmpty: false,
            exclude: []
        },
        number: {
            removeZero: false,
            exclude: []
        }
    };

    return merge(defaults, config);
}