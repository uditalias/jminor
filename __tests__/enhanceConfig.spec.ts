'use strict';

import enhanceConfig from "../lib/minifier/enhanceConfig";

describe('enhanceConfig', () => {

    it('should return default config', () => {

        expect(enhanceConfig({})).toMatchSnapshot();

    });

    it('should extend exclude array', () => {
        expect(enhanceConfig({
            null: {
                removeNull: true,
                exclude: ["first"]
            }
        })).toMatchSnapshot();
    });

});