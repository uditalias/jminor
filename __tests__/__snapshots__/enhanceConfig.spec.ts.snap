// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enhanceConfig should extend exclude array 1`] = `
Object {
  "array": Object {
    "exclude": Array [],
    "removeEmpty": false,
  },
  "boolean": Object {
    "exclude": Array [],
    "removeFalse": false,
  },
  "null": Object {
    "exclude": Array [
      "first",
    ],
    "removeNull": true,
  },
  "number": Object {
    "exclude": Array [],
    "removeZero": false,
  },
  "object": Object {
    "exclude": Array [],
    "removeEmpty": false,
  },
  "string": Object {
    "exclude": Array [],
    "removeEmpty": false,
  },
  "translateKeys": true,
}
`;

exports[`enhanceConfig should return default config 1`] = `
Object {
  "array": Object {
    "exclude": Array [],
    "removeEmpty": false,
  },
  "boolean": Object {
    "exclude": Array [],
    "removeFalse": false,
  },
  "null": Object {
    "exclude": Array [],
    "removeNull": false,
  },
  "number": Object {
    "exclude": Array [],
    "removeZero": false,
  },
  "object": Object {
    "exclude": Array [],
    "removeEmpty": false,
  },
  "string": Object {
    "exclude": Array [],
    "removeEmpty": false,
  },
  "translateKeys": true,
}
`;
